ctl.c

```diff
@@ -0,0 +1,78 @@
+
+#include <errno.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/param.h>
+#include <unistd.h>
+
+#include "ipc.h"
+
+static void usage(const char *progname) {
+	fprintf(stderr, "Usage: %s [command]\n"
+			"Accepted commands:\n"
+			"  reload - reload the config file\n",
+			progname);
+}
+
+static long reload_callback(VarlinkConnection *connection, const char *error,
+		VarlinkObject *parameters, uint64_t flags, void *userdata) {
+	bool *done = userdata;
+	*done = true;
+	return 0;
+}
+
+#define ARRAY_LENGTH(_arr) (sizeof(_arr) / sizeof(*(_arr)))
+
+int wait_for_event(VarlinkConnection *connection, bool *done) {
+	struct pollfd fds[1];
```

Comment:
Do we really need `poll` here? This utility is never going to do anything besides calling varlink functions, so I think blocking is fine.

Reply:
We don't, I managed to remove it by unsetting O_NONBLOCK.